src/ui/include/EditorNS/editor.h

```diff
@@ -317,47 +550,72 @@ namespace EditorNS
         void getSelectedTexts(std::function<void(const QStringList&)> callback);
 
         /**
-         * @brief Get the current editor language.
-         * @param callback Lambda or functor.
-         *                 Must accept QString type as parameter.
-         */
-        void getLanguage(std::function<void(const QString&)> callback); 
-        /**
          * @brief Get the currently selected text, or the word under the cursor
          *        if no text is selected.
          * @param callback Lambda or functor.
          *                 Must accept QStringList type as parameter.
          */
+
         void getCurrentWordOrSelections(std::function<void(const QStringList&)> callback); 
 
+        /**
+         * @brief Get the current content of the editor.
+         * @param callback Lambda or functor.
+         *        Must accept QString as type parameter.
+         */
+        void getValue(std::function<void(const QString&)> callback);
+
+        /**
+         * @brief Initialize the language cache for fast access.
+         */
+        static void initLanguageCache();
+
+
+        /**
+         * @brief  Workaround. Using isClean() right after creating an Editor always returns true.
+         *         This value gets set to 'true' if the Editor starts out as a newly created empty
+         *         document, and is 'false' if the document was loaded from any file.
+         * @return
+         */
+        bool isInitiallyBlank() const { return m_initiallyBlank; }
+        void setInitiallyBlank(bool blank) { m_initiallyBlank = blank; }
```

Comment:
Right now `DocEngine` closes an empty tab when opening a new tab if that empty tab was the only open tab. That is reasonably but goes wrong phenomenally with the new asynchronous Javascript. Reason for that is that when `DocEngine` loads several documents at once (like it does when restoring tabs at startup) it'll check whether an `Editor` is empty right after creating it. At that point the JavaScript hasn't completely initialized yet and `isClear()` returns its default value of `true`.

So a way was needed to determine whether an `Editor` was created as an empty document (in which case it can be closed by `DocEngine`) or not. `isInitiallyBlank` does that.